### PR TITLE
Remove pass-through activator for pages

### PR DESCRIPTION
**Pull request by rynowak**
This change removes a 'pass through' IRazorPageActivator that we were
using to avoid the default one initializing the Page twice.

The fix is to add an adapter so that the IRazorPage that the RazorView has
isn't the *real* page, it's the adapter. The adapter doesn't have anything
interesting to activate :).

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.Razor/RazorPageBase.cs [hunk 1]
```diff
@@ -755,7 +755,7 @@ private struct TagHelperAttributeInfo
             public bool Suppressed { get; set; }
         }
 
-        protected struct TagHelperScopeInfo
+        private struct TagHelperScopeInfo
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Razor/RazorPageBase.cs]: This isn't used anywhere else

File: src/Microsoft.AspNetCore.Mvc.RazorPages/DependencyInjection/MvcRazorPagesMvcCoreBuilderExtensions.cs [hunk 2]
```diff
@@ -114,7 +114,7 @@ internal static void AddServices(IServiceCollection services)
             services.TryAddSingleton<IPageModelFactoryProvider, DefaultPageModelFactoryProvider>();
 
             services.TryAddSingleton<IPageActivatorProvider, DefaultPageActivatorProvider>();
-            services.TryAddSingleton<IPageFactoryProvider, DefaultPageFactory>();
+            services.TryAddSingleton<IPageFactoryProvider, DefaultPageFactoryProvider>();
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.RazorPages/DependencyInjection/MvcRazorPagesMvcCoreBuilderExtensions.cs]: This was named wrong

File: src/Microsoft.AspNetCore.Mvc.RazorPages/PageBase.cs [hunk 3]
```diff
@@ -25,7 +25,7 @@ namespace Microsoft.AspNetCore.Mvc.RazorPages
     /// A base class for a Razor page.
     /// </summary>
     [PagesBaseClass]
-    public abstract class PageBase : RazorPageBase, IRazorPage
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.RazorPages/PageBase.cs]: This interface was redundant with the base-class

**Comment by pranavkm** [comment]
:shipit: as long as things compile.

**Pull request merged**